google/cloud/spanner/samples/samples.cc

```diff
@@ -3728,21 +3728,27 @@ void RunAllSlowInstanceTests(
 
   if (!run_slow_instance_tests) return;
 
+  auto const prod_service = google::cloud::internal::GetEnv(
+                                "GOOGLE_CLOUD_CPP_SPANNER_DEFAULT_ENDPOINT")
+                                .value_or("")
+                                .empty();
   auto const config_id =
       Basename(google::cloud::spanner_testing::PickInstanceConfig(
           google::cloud::Project(project_id), generator,
-          [](google::spanner::admin::instance::v1::InstanceConfig const&
-                 config) {
+          [prod_service](
```

Comment:
nit: maybe time to move the functor out?

Reply:
OK, done.